src/syntax/FStar.Syntax.Syntax.fsti

```diff
@@ -620,6 +621,7 @@ type sig_metadata = {
     sigmeta_fact_db_ids:list string;
     sigmeta_admit:bool; //An internal flag to record that a sigelt's SMT proof should be admitted
                         //Used in DM4Free
+    sigmeta_extension_data: list (string & dyn) //each extension can register some data with a sig
```

Comment:
In the extraction case, this would be a singleton list, right? Is there any other usage where this would be non-singleton?

Reply:
So far, yes, this is singleton when used in extraction. But, I could imagine that different phases could stash different information in here, e.g, some extension data is added after tosyntax, some after typechecking, etc. Also, could one conceivably have multiple extensions process a single sigelt ... maybe?